test/wasm_worker/proxy_abort.c

```diff
@@ -0,0 +1,44 @@
+#include <emscripten.h>
+#include <emscripten/wasm_worker.h>
+#include <assert.h>
+
+// Tests behavior of __proxy: 'abort' and __proxy: 'abort_debug' in Wasm Workers.
+
+void proxied_js_function(void);
+
+int might_throw(void(*func)()) {
+  int threw = EM_ASM_INT({
+    // Patch over assert() so that it does not abort execution on assert failure, but instead
+    // throws a catchable exception.
+    assert = function(condition, text) {
+      if (!condition) {
+        throw 'Assertion failed' + (text ? ": " + text : "");
+      }
+    };
+
+    try {
+      getWasmTableEntry($0)();
+    } catch(e) {
+      console.error('Threw an exception: ' + e);
+      return e.toString().includes('this function has been declared to only be callable from the main browser thread');
+    }
+    console.error('Function did not throw');
+    return 0;
+  }, func);
+  return threw;
+}
+
+void test() {
+  proxied_js_function();
+}
+
+void worker_main() {
+  REPORT_RESULT(might_throw(test));
```

Comment:
Can you not just do `might_throw(proxied_js_function)` here?  (i.e. do we need the test function at all?)

Reply:
I just mean the `test` wrapper function:

```
void test() {
  proxied_js_function();
}
```

It seems like a pointless wrapper around `proxied_js_function`?   Is it needed for some reason?  Why can't the address of `proxied_js_function` be used in place of the address of `test`?

Reply:
Yeah, that can be removed.